samcli/commands/validate/validate.py

```diff
@@ -95,3 +114,40 @@ def _read_sam_file(template):
         sam_template = yaml_parse(sam_template.read())
 
     return sam_template
+
+def _lint(ctx, template):
+    """
+    Parses provided SAM template and maps errors from CloudFormation template back to SAM template.
+
+    Parameters
+    -----------
+    ctx
+        Click context object
+    template
+        Path to the template file
+
+    """
+
+    try:
+        if ctx.debug:
+            (args, filenames, formatter) = cfnlint.core.get_args_filenames([template, "--debug"])
+        else:
+            (args, filenames, formatter) = cfnlint.core.get_args_filenames([template])
+        LOGGER.setLevel(logging.WARNING)
+        matches = list(cfnlint.core.get_matches(filenames, args))
+        if not matches:
+            click.secho("{} is a valid SAM Template".format(template), fg="green")
+        rules = cfnlint.core.get_used_rules()
+        matches_output = formatter.print_matches(matches, rules, filenames)
+
+        if matches_output:
+            if args.output_file:
```

Comment:
Where this `output_file` comes from and how users can set it?

Reply:
This is an cfn-lint optional parameter which we cannot specify for sam validate so it will never output to this file. Will remove in next commit. Optionally we can have it always print to an output file but we would need to have that be explicitly stated to the customer